forecasting/tests/unittests/test_ts_dataset.py

```diff
@@ -243,5 +243,47 @@ def test_subsequence(start_timestamp, end_timestamp, item_ids, datetimes, target
     pd.testing.assert_frame_equal(new_tsdf, ts_df)
 
 
-if __name__ == "__main__":
-    test_from_gluonts_list_dataset()
+@pytest.mark.parametrize(
+    "timestamps, expected_freq",
+    [
+        (["2020-01-01 00:00:00", "2020-01-02 00:00:00", "2020-01-03 00:00:00"], "D"),
+        (["2020-01-01 00:00:00", "2020-01-03 00:00:00", "2020-01-05 00:00:00"], "2D"),
+        (["2020-01-01 00:00:00", "2020-01-01 00:01:00", "2020-01-01 00:02:00"], "T"),
+        (["2020-01-01 00:00:00", "2020-01-01 01:00:00", "2020-01-01 02:00:00"], "H"),
+    ],
+)
+def test_when_dataset_constructed_from_dataframe_without_freq_then_freq_is_inferred(
```

Comment:
What about a test for an example that is invalid (not consistent frequency)? Such as:

["2020-01-01 00:00:00", "2020-01-02 00:00:00", "2020-01-04 00:00:00"]

Maybe this already exists. If so no action is required.

Reply:
in GluonTS we always assume uniformly sampled (perfectly regular) time series, with an explicit NaN if the time step is missing. We are (somewhat implicitly) assuming this for our data frames as well.

Reply:
This still doesn't explain what happens when it isn't uniformly sampled (should we throw an exception, and is there a test that covers this situation?)

Reply:
Added functionality to validate and added test.